experimental/web_dashboard/lib/src/pages/entries.dart

```diff
@@ -0,0 +1,163 @@
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:web_dashboard/src/api/api.dart';
+import 'package:intl/intl.dart' as intl;
+import 'package:web_dashboard/src/widgets/dialogs.dart';
+import 'package:web_dashboard/src/widgets/categories_dropdown.dart';
+
+import '../app.dart';
+
+class EntriesPage extends StatefulWidget {
+  @override
+  _EntriesPageState createState() => _EntriesPageState();
+}
+
+class _EntriesPageState extends State<EntriesPage> {
+  Category _selected;
+
+  @override
+  Widget build(BuildContext context) {
+    var appState = Provider.of<AppState>(context);
+    return Column(
+      children: [
+        CategoryDropdown(
+            api: appState.api.categories,
+            onSelected: (category) => setState(() => _selected = category)),
+        Expanded(
+          child: _selected == null
+              ? Center(child: CircularProgressIndicator())
+              : EntriesList(
+                  category: _selected,
+                  api: appState.api.entries,
+                ),
+        ),
+      ],
+    );
+  }
+}
+
+class EntriesList extends StatefulWidget {
+  final Category category;
+  final EntryApi api;
+
+  EntriesList({
+    @required this.category,
+    @required this.api,
+  }) : super(key: ValueKey(category.id));
+
+  @override
+  _EntriesListState createState() => _EntriesListState();
+}
+
+class _EntriesListState extends State<EntriesList> {
+  @override
+  Widget build(BuildContext context) {
+    if (widget.category == null) {
+      return _buildLoadingIndicator();
+    }
+
+    return FutureBuilder<List<Entry>>(
+      future: widget.api.list(widget.category.id),
+      builder: (context, futureSnapshot) {
+        if (!futureSnapshot.hasData) {
+          return _buildLoadingIndicator();
+        }
+        return StreamBuilder<List<Entry>>(
+          initialData: futureSnapshot.data,
+          stream: widget.api.subscribe(widget.category.id),
+          builder: (context, snapshot) {
+            if (!snapshot.hasData) {
+              return _buildLoadingIndicator();
+            }
+            return ListView.builder(
+              itemBuilder: (context, index) {
+                return EntryTile(
+                  category: widget.category,
+                  entry: snapshot.data[index],
+                );
+              },
+              itemCount: snapshot.data.length,
+            );
+          },
+        );
+      },
+    );
+  }
+
+  Widget _buildLoadingIndicator() {
+    return Center(child: CircularProgressIndicator());
+  }
+}
+
+class EntryTile extends StatelessWidget {
+  final Category category;
+  final Entry entry;
+
+  EntryTile({
+    this.category,
+    this.entry,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      title: Text(entry.value.toString()),
```

Comment:
nit: I've been told by folks on the Dart team that string interpolation is preferred to `toString`:

```
      title: Text('${entry.value}');
```

Reply:
You're right, I think it's [in the style guide](https://dart.dev/guides/language/effective-dart/usage#prefer-using-interpolation-to-compose-strings-and-values)